components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/AnalyticsUtils.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    public static boolean isThrottledFaultRequest(int errorCode) {
+        return errorCode >= Constants.ERROR_CODE_RANGES.THROTTLED_FAILURE_START
+                && errorCode < Constants.ERROR_CODE_RANGES.THROTTLED_FAILURE__END;
+    }
+
+    public static boolean isTargetFaultRequest(int errorCode) {
+        return errorCode >= Constants.ERROR_CODE_RANGES.TARGET_FAILURE_START
+                && errorCode < Constants.ERROR_CODE_RANGES.TARGET_FAILURE__END;
+    }
+
+    public static FAULT_EVENT_TYPE getFaultyType(MessageContext messageContext) {
+        int errorCode = (int) messageContext.getProperty(SynapseConstants.ERROR_CODE);
+        if (isAuthFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.AUTH;
+        } else if (isThrottledFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.THROTTLED;
+        } else if (isTargetFaultRequest(errorCode)) {
+            return FAULT_EVENT_TYPE.TARGET;
+        } else {
+            return FAULT_EVENT_TYPE.OTHER;
+        }
+    }
+
+    public static boolean isFaultRequest(MessageContext messageContext) {
+        return messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE);
+    }
+
+    public static boolean isSuccessRequest(MessageContext messageContext) {
+        return !messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE)
+                && APISecurityUtils.getAuthenticationContext(messageContext) != null;
+    }
+
+    public static boolean isResourceNotFound(MessageContext messageContext) {
+        if (messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE)) {
+            int errorCode = (int) messageContext.getProperty(SynapseConstants.ERROR_CODE);
+            return messageContext.getPropertyKeySet().contains(RESTConstants.PROCESSED_API)
+                    && errorCode == Constants.RESOURCE_NOT_FOUND_ERROR_CODE;
+        }
+        return false;
+    }
+
+    public static boolean isMethodNotAllowed(MessageContext messageContext) {
+        if (messageContext.getPropertyKeySet().contains(SynapseConstants.ERROR_CODE)) {
+            int errorCode = (int) messageContext.getProperty(SynapseConstants.ERROR_CODE);
+            return messageContext.getPropertyKeySet().contains(RESTConstants.PROCESSED_API)
+                    && errorCode == Constants.METHOD_NOT_ALLOWED_ERROR_CODE;
+        }
+        return false;
+    }
+
+    public static boolean isAPINotFound(MessageContext messageContext) {
+        return !messageContext.getPropertyKeySet().contains(RESTConstants.PROCESSED_API);
+    }
+
+    public static boolean isProxyAPI(MessageContext messageContext) {
+        return messageContext.getPropertyKeySet().contains(RESTConstants.PROCESSED_API);
+    }
+
+    public static EVENT_TYPE getEventType(MessageContext messageContext) {
+        if (isSuccessRequest(messageContext)) {
+            return EVENT_TYPE.SUCCESS;
+        } else if (isFaultRequest(messageContext)) {
```

Comment:
isn't a fault request should be one of the known types or "OTHER"? when does this return true?

Reply:
method `getEventType` is not used. So I'll remove it.

Reply:
Fixed